libraries/AP_Motors/AP_MotorsHeli_RSC.cpp

```diff
@@ -305,46 +312,32 @@ void AP_MotorsHeli_RSC::output(RotorControlState state)
         governor_reset();
         _autothrottle = false;
         _governor_fault = false;
-        if (_in_autorotation) {
-            // if in autorotation, set the output to idle for autorotation. This will tell an external governor to use fast ramp for spool up.
-            // if autorotation idle is set to zero then default to the RSC idle value.
-            if (_arot_idle_output == 0) {
+
+        // check if we need to use autorotation idle throttle
+        if (autorotation.get_idle_throttle(_idle_throttle)) {
+            // if we are here then we are autorotating
```

Comment:
@MattKear this guarantees the aircraft is in autorotation even if the throttle idle value is zero. I guess if you’re in our autorotation, it returns a true value for this call?

Reply:
If you are in the autorotation mode the function autorotation.get_idle_throttle() populates the _idle_throttle variable with the value set by H_RSC_AROT_IDLE and returns true.  This is then assigned to the control output and returns early.  if we are not in the autorotation then _idle_throttle is not modified and the autorotation.get_idle_throttle() returns false giving access to the subsequent cases.
In the case of both turbine and piston, they would have to set H_RSC_AROT_IDLE to match H_RSC_IDLE to maintain the engine at idle during the autorotation.

Reply:
@MattKear I fixed the cooldown feature in this PR.  it required that I change the functionality of get_idle_throttle(). One because when AROT_IDLE was zero then it would use the cooldown idle.  and two, for some reason it would always use the cooldown throttle even when I used the autorotation state.  Look at what I did and let me know what  you think.
@Ferruccio1984 checkout and test the changes.  I think that works as it was meant to work.